netbox/ipam/views.py

```diff
@@ -333,7 +333,14 @@ def get(self, request, pk):
         ).annotate_depth(
             limit=0
         )
-        child_prefixes = add_available_prefixes(aggregate.prefix, child_prefixes)
+
+        # Update the ipam_show_available cookie if request specifies it
+        if request.GET.get('show_available') is not None:
+            request.session['ipam_show_available'] = request.GET.get('show_available') == 'true'
```

Comment:
We need to put some development guidance in place before we start storing session data, specifically around naming convention and user experience expectations. (We've somehow avoided it thus far.) I'll raise it with the other maintainers to see what seems prudent. IMO we should hold off on leveraging session state until we have a mechanism for persistence across logins (see #3294).

Reply:
Yeah, I was just as surprised when I noticed that we've not been explicitly dealing with cookies before (ignoring the Django-based ones).

Reply:
Should settings like these (i.e. visual) exist for users in the first place? Someone that is not logged in might also be interested in toggling the view.

With that said, I do agree that we need a developer interface for abstracting cookies and how they're handled, though I don't think it should rely on the `users` model.